apps/prairielearn/assets/scripts/examTimeLimitCountdown.ts

```diff
@@ -0,0 +1,50 @@
+import { Countdown } from './lib/countdown';
+import { saveQuestionFormData } from './lib/confirmOnUnload';
+import { onDocumentReady, decodeData, parseHTMLElement } from '@prairielearn/browser-utils';
+import { html } from '@prairielearn/html';
+
+onDocumentReady(() => {
+  const timeLimitData = decodeData<{
+    serverRemainingMS: number;
+    serverTimeLimitMS: number;
+    serverUpdateURL: string;
+    canTriggerFinish: boolean;
+    csrfToken: string;
+  }>('time-limit-data');
+  new Countdown(
```

Comment:
In PT, we typically use functions instead of classes for this kind of thing (e.g. `setupCountdown`. And rather than having 8 arguments, we'll typically use some kind of "options" object with named keys, which is easier to reason about:

```ts
setupCountdown({
  displaySelector: '#countdownDisplay',
  progressSelector: '#countdownProgress',
  // ...
});

Reply:
> In PT, we typically use functions instead of classes for this kind of thing (e.g. `setupCountdown`.

That would require a bunch of inner functions, as there's a need to store internal data from the setup to the callbacks. Is that ok? I thought the class was a bit more legible in that regard.

Reply:
Nested functions are fine, yeah; we do that extensively in PT too. We typically use classes only where we can actually derive value from inheritance, abstract methods, and other class-specific things (and we do not frequently use those anything that's class-specific). Another smell for me is calling `new Foo()` without saving the instance or subsequently doing anything with it.

Reply:
Done in a3e304837f76f633d1a4efed2eef4d33d1cb38e5.